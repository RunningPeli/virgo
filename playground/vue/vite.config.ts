import VirgoResolver from '@runningpeli/virgo/resolver'
import tailwindcss from '@tailwindcss/vite'
import vue from '@vitejs/plugin-vue'
import components from 'unplugin-vue-components/vite'
import { defineConfig } from 'vite'

// https://vite.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    tailwindcss(),
    components({
      dts: true,
      resolvers: [
        VirgoResolver(),
      ],
    }),
  ],
})
