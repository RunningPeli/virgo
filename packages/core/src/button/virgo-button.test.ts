import { fireEvent, render, screen } from '@testing-library/vue'
import { describe, expect, it } from 'vitest'
import VirgoButton from './virgo-button.vue'

describe('button', () => {
  it('should emit a click event when clicked', async () => {
    const { emitted } = render(VirgoButton, {
      props: {
        text: 'Hello',
      },
    })

    await fireEvent.click(screen.getByText('Hello'))

    expect(emitted()).toMatchInlineSnapshot(`
      {
        "click": [
          [
            MouseEvent {
              "isTrusted": false,
            },
          ],
        ],
      }
    `)
  })

  describe('hello button', () => {
    it('should render a button with hello text', () => {
      const button = render(VirgoButton, {
        props: {
          text: 'Hello'
        },
      })

      expect(button.html()).toMatchSnapshot()
    })
  })
})