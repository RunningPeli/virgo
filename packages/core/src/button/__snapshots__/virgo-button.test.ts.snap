// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button > hello button > should render a button with hello text 1`] = `"<button>Hello</button>"`;
